integration/testrunner/http.go

```diff
@@ -64,18 +74,64 @@ func testHTTPEndpoint(ctx context.Context, tb testing.TB, endpointURL, idToken,
 	}); err != nil {
 		tb.Fatal(err)
 	}
-
 	bqClient := makeBigQueryClient(ctx, tb, projectID)
-
+	time.Sleep(10 * time.Second)
 	bqQuery := makeQueryForHTTP(*bqClient, id, projectID, datasetQuery)
-	queryIfAuditLogExistsWithRetries(ctx, tb, bqQuery, cfg, "httpEndpointTest")
+	tb.Log("querying bigquery table using the following statement:")
+	tb.Log(bqQuery.Q)
+	results := queryAuditLogsWithRetries(ctx, tb, bqQuery, cfg, "httpEndpointTest")
+	wantNum := 1
+	if len(results) != wantNum {
+		tb.Errorf("log number doesn't match (-want +got):\n - %d\n + %d\n", wantNum, len(results))
+	} else {
+		diffLogEntry(tb, results[0], isHTTP("HTTP"), 0)
+	}
 }
 
 func makeQueryForHTTP(client bigquery.Client, id, projectID, datasetQuery string) *bigquery.Query {
-	// Cast to int64 because the result checker expects a number.
-	queryString := fmt.Sprintf("SELECT CAST(EXISTS (SELECT * FROM `%s.%s` WHERE labels.trace_id=?", projectID, datasetQuery)
-	queryString += ` AND jsonPayload.service_name IS NOT NULL`
-	queryString += ` AND jsonPayload.authentication_info.principal_email IS NOT NULL`
-	queryString += ") AS INT64)"
+	queryString := fmt.Sprintf(` WITH temptable AS (
+		SELECT *
+		FROM %s.%s
+		WHERE labels.trace_id=?
+ 	)
+ 	SELECT TO_JSON(t) as result FROM temptable as t
+	`, projectID, datasetQuery)
 	return makeQuery(client, id, queryString)
 }
+
+func diffLogEntry(tb testing.TB, logEntry *logpb.LogEntry, isHTTPService bool, index int) {
```

Comment:
This is no longer making sense. There are too much if/else and individual field comparisons. I can see why comparing the whole LogEntry is challenging, which is why I think we should switch gear to focus on the critical information in the logs - "_when_ and _who_ called _which_ application and accessed _what_ data. And _why_ the access was necessary".

The basic shared checks, e.g.

- The log name is set
- When - make sure timestamp is setup
- Who - make sure principalEmail is set; we probably want to roughly check if it's an valid email
- Which - make sure serviceName is set; it will vary across apps
- What - (we didn't do that consistently with gRPC and HTTP so we can skip for now)
- Why - make sure justification fields are set with the right value. This is only applicable for gRPC at the moment.

For "what" and "why", the behavior will vary, which we should fix **separately**. And once we have that consistency, it'd become easier for us to revisit whether we want to compare the whole LogEntry.

Reply:
There is some bugs in the src code to set up the timestamp, see https://github.com/abcxyz/lumberjack/actions/runs/4168063607/jobs/7214517647, I checked the logs, no timestamp were set for some logs. I will fix it later in another PR.

Reply:
In addition, which field do you mean by `what`, `jsonPayload.methodName` or `jsonPayload.resourceName`

Reply:
"What" is usually `jsonPayload.resourceName`.